### ISSUE TITLE
Unable to missed_cache_period_seconds to 0 on remote repo during update

### ISSUE THREAD

**Issue by jamestoyer** [jfrog/terraform-provider-artifactory#217]
**Describe the bug**
**Note:** This bug relates to updating a `artifactory_remote_repository` and not creation of a `artifactory_remote_repository` See Additional Context as to why it doesn't apply to creates.

When using the `artifactory_remote_repository` resource, setting `missed_cache_period_seconds` to 0 fails to persist the change on update. Each subsequent run will then attempt to set it to zero but it is not persisted in Artifactory.
```
missed_cache_period_seconds : 1800 change to 0
```

This works on v2.2.15 and earlier but fails on v2.3.16. Both providers were used to test against the same Artifactory version, 7.27.10, and were tested using Terraform 1.0.6 and 1.0.11.

Example TF:
```hcl
resource "artifactory_remote_repository" "this" {
  key                         = "remote-repo"
  package_type                = "npm"
  url                         = "https://registry.npmjs.org/"
  repo_layout_ref             = "npm-default"
  propagate_query_params      = true
  missed_cache_period_seconds = 0
}


```

**Requirements for and issue**
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [x] Your version of terraform

**Expected behavior**
When updating a repository to have a `missed_cache_period_seconds` of 0 the change should be correctly sent to Artifactory and persisted. 

**Additional context**
This comes about as creating a remote repository with a `missed_cache_period_seconds` of 0 fails due to a bug with the Artifactory API and not the provider. I've validated in the past both the Artifactory v6 and v7 APIs just straight up fail to do the right thing on create when default values are set. It treats them as not being set.

**Comment by chb0github** [comment]
When I run the initial code for setup:

```hcl
resource "artifactory_remote_repository" "this" {
  key                         = "remote-repo-npm"
  package_type                = "npm"
  url                         = "https://registry.npmjs.org/"
  repo_layout_ref             = "npm-default"
  propagate_query_params      = true
  missed_cache_period_seconds = 1800
}
```

I get:
```
Error: cannot use propagate_query_params with repository type npm. This parameter can be used only with generic repositories

  on sample.tf line 11, in resource "artifactory_remote_repository" "this":
  11: resource "artifactory_remote_repository" "this" {
```

Once I remove that, I get:

```
Xray and projects functionality will not work with any auth method other than
access tokens (Bearer)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

artifactory_remote_repository.this: Creating...
artifactory_remote_repository.this: Creation complete after 1s [id=remote-repo-npm]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
christianb@christianb-mac terraform-provider-artifactory % 
```

verifying:
```bash
christianb@christianb-mac terraform-provider-artifactory % curl  http://localhost:8081/artifactory/api/repositories/remote-repo-npm | jq .missedRetrievalCachePeriodSecs
1800
```

The problem is now fixed and, it was/is because of how go and it's json processor treat `0`. This will be released in 2.6.19

BTW: We've been moving to a new model of having dedicated resource types - the generic version you use now is slowly going away; you should try using this:

```hcl
resource "artifactory_remote_npm_repository" "thing" {
  key                         = "remote-thing-npm"
  url                         = "https://registry.npmjs.org/"
  repo_layout_ref             = "npm-default"
  missed_cache_period_seconds = 0
  list_remote_folder_items    = true
  mismatching_mime_types_override_list = "application/json,application/xml"
}
```

**Comment by jamestoyer** [comment]
> BTW: We've been moving to a new model of having dedicated resource types - the generic version you use now is slowly going away; you should try using this:

We'll need some time to look into this. We currently have several hundred repos managed by this resource, of different types, so that'll take time. I only use NPM in this example, but I suspect it's an issue for all repo types that use it. 

I'm hoping the `moved` block in TF 1.1 will help us migrate when that drops.
